src/System.Windows.Forms.Primitives/src/Interop/Interop.PARAM.cs

```diff
@@ -15,13 +15,12 @@ internal static class PARAM
         public static IntPtr FromLowHigh(int low, int high)
             => (IntPtr)ToInt(low, high);
 
-        public static unsafe IntPtr FromLowHighUnsigned(int low, int high)
+        public static IntPtr FromLowHighUnsigned(int low, int high)
             // Convert the int to an uint before converting it to a pointer type,
             // which ensures the high dword being zero for 64-bit pointers.
             // This corresponds to the logic of the MAKELPARAM/MAKEWPARAM/MAKELRESULT
             // macros.
-            // TODO: Use nint (with 'unchecked') instead of void* when it is available.
-            => (IntPtr)(void*)unchecked((uint)ToInt(low, high));
+            => unchecked((nint)(nuint)(uint)ToInt(low, high));
```

Comment:
I believe the following should be sufficient:
```suggestion
            => unchecked((nint)(uint)ToInt(low, high));
```

Due to converting from `uint` to `nint`, C# will just emit a `conv.u` which is a zero extension: <https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoHAZgAJ9KBhSgb1MpeqoEkA7GABRgUoBZAgAo03SgDcIGAK5wAlI2atVOAOyVZnAMYALODoDWcACYiRncTAWXZ12/e4LpcxQG4VLAL5fKfikouXn4hPDEJV3klJhJVNU1tfUMTc0sHEScbKI8/XxJvIA>

Reply:
@tannergooding well, its not readable, you don't know whether `(nint)(uint)` means `uint -> int -> nint` or `uint -> nuint -> nint` - see [discussion above](https://github.com/dotnet/winforms/pull/3603#discussion_r461491609) - and in fact for constants Roslyn was not doig zero-extension for a while.

Reply:
It should be no different, logically, from doing a cast of `uint` to `int` (if a 32-bit process) or `uint` to `long` (if a 64-bit process), I don't think you would write `(long)(ulong)(uint)value` or `(int)(uint)(uint)value` so I (personally) wouldn't expect to see `(nint)(nuint)(uint)value` here either.

> and in fact for constants Roslyn was not doing zero-extension for a while.

AFAIK, all these cases have been fixed. There were a few edge cases in the initial preview, and all of those that were made known have been fixed.
Going through a few cases, I didn't see any issues with constants when comparing `nint`/`nuint` to `int`/`uint` in RC2 (I did not check if they had all made it into RC1).

Reply:
My point is that its not readable because the operation does two things at a time, changing signed-ness and size of the number, and different programming languages spec differently what happens. C# has (apparently) specced it to zero extend, i.e. first widen _then_ change sign. This is not obvious when reading the source, it could just as well first change sign then widen, sign-extending in the process.